geoh5py/objects/surveys/electromagnetics/base.py

```diff
@@ -219,55 +231,61 @@ def copy(  # pylint: disable=too-many-arguments
         if parent is None:
             parent = self.parent
 
-        omit_list = [
-            "_metadata",
-            "_receivers",
-            "_transmitters",
-            "_base_stations",
-            "_tx_id_property",
-        ]
-        kwargs["omit_list"] = omit_list
-        metadata = self.metadata.copy()
         new_entity = super().copy(
             parent=parent,
             clear_cache=clear_cache,
             copy_children=copy_children,
             mask=mask,
             cell_mask=cell_mask,
+            omit_list=OMIT_LIST,
             **kwargs,
         )
 
-        metadata["EM Dataset"][new_entity.type] = new_entity.uid
+        for key, value in self.metadata["EM Dataset"].items():
+            if not isinstance(value, (uuid.UUID, type(None))):
+                new_entity.edit_metadata({key: value})
```

Comment:
**suggestion (maintanability):** add some comment

quoting you 😉 
> UUIDs will get re-set when the copy of the transmitter gets set, then the metadata gets updates

Reply:
Done